### ISSUE TITLE
Error Foam Simulation from Tutorial

### ISSUE THREAD

**Issue by Tiiri** [EPhysLab-UVigo/VisualSPHysics#4]
I followed the tutorial and until the foam simulation everything worked fine.
But when I use the settings from the tutorial I can't run the simulation:
![physics_settings](https://user-images.githubusercontent.com/44700270/49714506-c4dc6d00-fc55-11e8-8b35-d475e0d3490f.PNG)
I am able to start it when I disable custom domain limits, but I get this error:
![simulation_error](https://user-images.githubusercontent.com/44700270/49714540-e4739580-fc55-11e8-87ee-76d9a3d8f107.PNG)

Thanks in advance for any advice.

**Comment by Rodland** [comment]
Hi Tiiri, It seems that you are loading the particles of the floating body instead of the fluid particles. Use the "PartFluid_" input prefix instead.

Best regards!

**Comment by Tiiri** [comment]
Hi Rodland,
thanks for the fast answer!
With "PartFluid_" I can't run the foam simulation at all.

Thanks for your help!

**Comment by Rodland** [comment]
You should also select the: `CaseFloating_out\CaseFloating.xml` instead of `CaseFLoating_out.xml`. The later has some constants missing!

**Comment by Tiiri** [comment]
I tried both. Doesn't change anything for the ability to run it.

What seems to enable the running is to leave Min X at 0 (with 2 the field turns red and run button is disabled - even when "enable custom domain limits" is unticked). 
But it generates empty .vtk.

**Comment by Rodland** [comment]
If the xml is loaded properly it is not necessary to enable the custom domain. Check the console to find some error.

**Comment by Tiiri** [comment]
As the console history is limited I can't see whether there was an error before the exports. But the part I see is without errors. But it doesn't get any fluid particles (if I interpret the output correctly) and on closer look I realised, that all the numbers are the same.
![foamsimulation_output](https://user-images.githubusercontent.com/44700270/49736393-b2345900-fc92-11e8-938b-67063339e0e6.PNG)

I will do the whole tutorial again tomorrow or on Wednesday and try again, just in case something went wrong along the way.
BTW in point 19 of the tutorial it is saying to set the custom domain limit.

Thank you very much for your help!

**Comment by Tiiri** [comment]
Okay, I tried the exact same files as yesterday again and it is working now :) I didn't change anything, who knows why ...
After I ran it once successfully I can change the custom domain limits, but not before.
Thank you for your help and this add-on!

P.S.: In step 18 of the tutorial I realised, that I can't see the number of particles like mentioned there. It is only shown two steps later while running the simulation. Otherwise very good and detailed tutorial!

**Comment by Tiiri** [comment]
Hi,
sorry for bothering you again.

I simulated the same example with a higher resolution (dp = "0.1"). Import and everything works fine now, but after the foam simulation blender crashes while showing later frames.
In the foam vtk files from these frames the first row(s) of points contain `-nan(ind)`. I suspect this as the reason for the crash. 
After rerunning the foam simulation the issue still occurs but not in the same frame.
vtk for last working frame:
[foam_0148.zip](https://github.com/EPhysLab-UVigo/VisualSPHysics/files/2671240/foam_0148.zip)
vtk for first frame that crashes blender:
[foam_0149.zip](https://github.com/EPhysLab-UVigo/VisualSPHysics/files/2671243/foam_0149.zip)
(I can't upload .vtk here so I zipped them.)
I can send you the complete output and blender file if needed.

Thanks for any help!
